src/placeos-frontend-loader/loader.cr

```diff
@@ -127,24 +148,18 @@ module PlaceOS::FrontendLoader
         unload(repository, content_directory)
       end
 
-      # Clone and pull the repository
-      clone_and_pull(
-        repository_folder_name: repository.folder_name,
-        repository_uri: repository.uri,
-        content_directory: content_directory,
-        username: username,
-        password: password,
-        branch: branch,
-      )
-
       hash = repository.should_pull? ? "HEAD" : repository.commit_hash
 
-      # Checkout repository to commit on the model
-      Git.checkout_branch(branch, repository.folder_name, content_directory)
-      Git._checkout(repository_directory, hash, raises: false)
+      # Download and extract the repository at given branch or commit
+      ref = Remote::Reference.new(repository.uri, branch: "master", hash: hash)
+
+      loaded_actioner = get_actioner(ref)
 
-      # Grab commit for the cloned/pulled repository
-      checked_out_commit = Git.current_repository_commit(repository.folder_name, content_directory)
+      # add to remote manger
+      loaded_actioner.download(ref: ref, hash: hash, branch: branch, path: repository_directory)
```

Comment:
You're passing the `remote` as a an argument, but fetching it from the class again here